File: streaming-server/src/main/java/io/github/hs96wings/streaming_server/chat/dto/ChatRoomListResDto.java
package io.github.hs96wings.streaming_server.chat.dto;

import io.github.hs96wings.streaming_server.chat.domain.ChatRoom;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ChatRoomListResDto {
    private Long roomId;
    private String roomName;

    public static ChatRoomListResDto from(ChatRoom chatRoom) {
        return ChatRoomListResDto.builder()
                .roomId(chatRoom.getId())
                .roomName(chatRoom.getName())
                .build();
    }
}

File: streaming-server/src/main/java/io/github/hs96wings/streaming_server/chat/dto/ChatRoomListResDto.kt
package io.github.hs96wings.streaming_server.chat.dto

import io.github.hs96wings.streaming_server.chat.domain.ChatRoom

data class ChatRoomListResDto (
    val roomId: Long,
    val roomName: String
) {
    companion object {
        @JvmStatic
        fun from(chatRoom: ChatRoom): ChatRoomListResDto {
            return ChatRoomListResDto(
                    roomId = chatRoom.id ?: throw IllegalStateException("chatRoom ID cannot be null"),
                    roomName = chatRoom.name
            )
        }
    }
}
